syntax = "proto2";

package clockwork;

enum RequestType {
  REQ_MODEL_LOAD = 1;
  REQ_MODEL_INFERENCE = 2;

  RES_MODEL_LOAD = 101;
  RES_MODEL_INFERENCE = 102;
}

message ModelLoadRequest {
  required uint32 model_id = 1;
  required uint32 batchsize = 2;
  required uint64 blob_a_len = 3;
  required uint64 blob_b_len = 4;
}

message ModelLoadResponse {
  required int32 status = 1;
}


message ModelInferenceRequest {
  required uint32 model_id = 1;
}

message ModelInferenceResponse {
  required int32 status = 1;
}
